pkg/githubclient/githubclient.go

```diff
@@ -43,6 +43,38 @@ func New(ctx context.Context, appID, rsaPrivateKeyPEM string) (*GitHub, error) {
 	}, nil
 }
 
+func NewWithPermissions(ctx context.Context, appID, gitHubInstallID, rsaPrivateKeyPEM string, permissions map[string]string, authOpts ...githubauth.Option) (*GitHub, error) {
```

Comment:
Given my above comment, we wouldnt need this function and the client can be created in the NewLogIngestor func.

Reply:
It's worth noting that `githubclient` is also used by the retry server ([example](https://github.com/abcxyz/github-metrics-aggregator/blob/main/pkg/retry/retry.go#L172)). Previously I had both the retry server and the artifact job using `func New` to create a client that was initialized with the right set of permissions for both, but iirc I was getting some error about the github app not having access to the permissions needed for `ListHookDeliveries` and `RedeliverHookDelivery`.

Reply:
I think I was aiming for something like this except generating a github rest client: https://github.com/abcxyz/github-metrics-aggregator/blob/main/pkg/review/job.go#L40-L60

Reply:
done